src/theory/quantifiers/bv_inverter.cpp

```diff
@@ -261,17 +262,107 @@ Node BvInverter::getPathToPv(Node lit, Node pv, Node sv, Node pvs,
   return slit;
 }
 
-Node BvInverter::solve_bv_constraint(Node sv,
-                                     Node sv_t,
-                                     Node t,
-                                     Kind rk,
-                                     bool pol,
-                                     std::vector<unsigned>& path,
-                                     BvInverterModelQuery* m,
-                                     BvInverterStatus& status) {
+Node BvInverter::solve_bv_lit(Node sv,
+                              Node lit,
+                              bool pol,
+                              std::vector<unsigned>& path,
+                              BvInverterModelQuery* m,
+                              BvInverterStatus& status) {
+
   unsigned index;
   unsigned nchildren;
   NodeManager* nm = NodeManager::currentNM();
+  Kind k;
+
+  do {
+    Assert(!path.empty());
+    index = path.back();
+    Assert(index < lit.getNumChildren());
+    path.pop_back();
+    k = lit.getKind();
+    if (k != NOT) break;
+    pol = !pol;
+    lit = lit[index];
+  } while (k == NOT);
+
```

Comment:
You don't need to have a while loop here I think, the rewriter will eliminate 2+ occurrences of NOT.